### PR TITLE
ER-850 - Fixing illegal 2019 wages

### PR DESCRIPTION
**Pull request by chrisjensenuk**

### CODE REVIEW

File: src/Data/Recruit.ChangeScripts/2018-12-18_020-fix_illegal_wages_2019.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        let d = hex.substr(16, 16);
+        hex = a + b + c + d;
+        let uuid = hex.substr(0, 8) + "-" + hex.substr(8, 4) + "-" + hex.substr(12, 4) + "-" + hex.substr(16, 4) + "-" + hex.substr(20, 12);
+        return uuid;
+    }
+
+    print("Start updating Vacancies with illegal 2019 wages.");
+
+    const query = {
+            "startDate": { $gte: ISODate("2019-04-01T00:00:00.000Z") },
```
Comment (sachpatel) [inline src/Data/Recruit.ChangeScripts/2018-12-18_020-fix_illegal_wages_2019.js]: Do we want to exclude `isDeleted` vacancies and those that are in `Draft` or `Closed` state?

File: src/Data/Recruit.ChangeScripts/2018-12-18_020-fix_illegal_wages_2019.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+            var perHourWage = doc.wage.fixedWageYearlyAmount / 52 / doc.wage.weeklyHours;
+            
+            if(perHourWage < apprenticeshipNationalMinimumWage2019){
+                print(`Found illegal 2019 wage. '${perHourWage}' per hour in document '${toGUID(doc._id.hex())}'`);
+
+                let writeResult = db.vacancies.update({
+                    "_id": doc._id
+                }, {
+                    $set: {
```
Comment (sachpatel) [inline src/Data/Recruit.ChangeScripts/2018-12-18_020-fix_illegal_wages_2019.js]:
```suggestion
$set: { "wage.wageType": "NationalMinimumWageForApprentices" },
$unset: { "wage.fixedWageYearlyAmount" : "" }
```

Comment (chrisjensenuk) [reply src/Data/Recruit.ChangeScripts/2018-12-18_020-fix_illegal_wages_2019.js]: Note to self: Remove the batching loop as I think its wrong and not needed

**Comment by chrisjensenuk** [comment]
Note to self: Remove the batching loop as I think its wrong and not needed

File: src/Data/Recruit.ChangeScripts/2018-12-18_020-fix_illegal_wages_2019.js (hunk 1 above)
Comment (chrisjensenuk) [reply src/Data/Recruit.ChangeScripts/2018-12-18_020-fix_illegal_wages_2019.js]: I think we should only change live vacancies as all others will be picked up through validation. Will change.

File: src/Data/Recruit.ChangeScripts/2018-12-18_020-fix_illegal_wages_2019.js (hunk 2 above)
Comment (chrisjensenuk) [reply src/Data/Recruit.ChangeScripts/2018-12-18_020-fix_illegal_wages_2019.js]: 👍

**Pull request merged**